python/dataproc_templates/gcs/gcs_to_bigquery.py

```diff
@@ -0,0 +1,115 @@
+        )
+        parser.add_argument(
+            f'--{constants.GCS_BQ_OUTPUT_DATASET}',
+            dest=constants.GCS_BQ_OUTPUT_DATASET,
+            required=True,
+            help='BigQuery dataset for the output table'
+        )
+        parser.add_argument(
+            f'--{constants.GCS_BQ_OUTPUT_TABLE}',
+            dest=constants.GCS_BQ_OUTPUT_TABLE,
+            required=True,
+            help='BigQuery output table name'
+        )
+        parser.add_argument(
+            f'--{constants.GCS_BQ_INPUT_FORMAT}',
+            dest=constants.GCS_BQ_INPUT_FORMAT,
+            required=True,
+            help='Input file format (one of: avro,parquet,csv)',
+            choices=[
+                constants.GCS_BQ_AVRO_FORMAT,
+                constants.GCS_BQ_PRQT_FORMAT,
+                constants.GCS_BQ_CSV_FORMAT
+            ]
+        )
+        parser.add_argument(
+            f'--{constants.GCS_BQ_LD_TEMP_BUCKET_NAME}',
+            dest=constants.GCS_BQ_LD_TEMP_BUCKET_NAME,
+            required=True,
+            help='Spark BigQuery connector temporary bucket'
+        )
+
+        known_args: argparse.Namespace
+        known_args, _ = parser.parse_known_args(args)
+
+        return vars(known_args)
+
+    def run(self, spark: SparkSession, args: Dict[str, Any]) -> None:
+
+        log4jLogger = spark.sparkContext._jvm.org.apache.log4j  # pylint: disable=protected-access,invalid-name
+        logger = log4jLogger.LogManager.getLogger(__name__)
+
+        # Arguments
+        input_file_location = args[constants.GCS_BQ_INPUT_LOCATION]
+        big_query_dataset = args[constants.GCS_BQ_OUTPUT_DATASET]
+        big_query_table = args[constants.GCS_BQ_OUTPUT_TABLE]
+        input_file_format = args[constants.GCS_BQ_INPUT_FORMAT]
+        bq_temp_bucket = args[constants.GCS_BQ_LD_TEMP_BUCKET_NAME]
+
+        logger.info(
+            "Starting GCS to Bigquery spark job with parameters:\n"
+            f"{pprint.pformat(args)}"
+        )
+
+        # Read
+        if input_file_format == constants.GCS_BQ_PRQT_FORMAT:
+            input_data = spark.read \
+                .parquet(input_file_location)
+        elif input_file_format == constants.GCS_BQ_AVRO_FORMAT:
+            input_data = spark.read \
+                .format(constants.GCS_BQ_AVRO_EXTD_FORMAT) \
+                .load(input_file_location)
+        elif input_file_format == constants.GCS_BQ_CSV_FORMAT:
+            input_data = spark.read \
+                .format(constants.GCS_BQ_CSV_FORMAT) \
+                .option(constants.GCS_BQ_CSV_HEADER, True) \
+                .option(constants.GCS_BQ_CSV_INFER_SCHEMA, True) \
+                .load(input_file_location)
+
+        # Write
+        input_data.write \
+            .format(constants.GCS_BQ_OUTPUT_FORMAT) \
+            .option(constants.GCS_BQ_OUTPUT, big_query_dataset + "." + big_query_table) \
+            .option(constants.GCS_BQ_TEMP_BUCKET, bq_temp_bucket) \
+            .mode("append") \
```

Comment:
This should e a template property. User should be able to specify the save mode (i.e. overwrite, append, errorifexists, ignore)

Reply:
Added! Now the save mode is an argument and it can be set to any of the 4 modes available. I set it to default to `append`, but we could make it mandatory. Thoughts?

One thing to note is that the Java version of this template also hard-codes `append`. Should we consider making it a parameter as well?

Reply:
Great!

imo it does not matter that much if it is mandatory or optional as long as the default value is clearly specified in the readme

per the java template, I would recommend creating a separate issue.